src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -118,35 +113,29 @@ public override string FullPath
             }
         }
 
-        private static string GetAssemblyNameFromPath(string path)
+        public override string Display
         {
-            // AssemblyName.GetAssemblyName(path) is not available on CoreCLR.
-            // Use our metadata reader to do the equivalent thing.
-            using (var reader = new PEReader(FileUtilities.OpenRead(path)))
+            get
             {
-                var metadataReader = reader.GetMetadataReader();
-                var assemblyDefinition = metadataReader.GetAssemblyDefinition();
-                return metadataReader.GetString(assemblyDefinition.Name);
+                if (_lazyDisplayName == null)
+                {
+                    _lazyDisplayName = Path.GetFileName(this.FullPath);
+                }
+
+                return _lazyDisplayName;
             }
         }
 
-        public override string Display
+        public override string Id
         {
             get
             {
-                if (_lazyDisplayName == null)
+                if (_lazyId == null)
                 {
-                    try
-                    {
-                        _lazyDisplayName = GetAssemblyNameFromPath(_fullPath);
-                    }
-                    catch (Exception)
-                    { }
-
-                    _lazyDisplayName = _lazyDisplayName ?? Path.GetFileName(this.FullPath);
+                    _lazyId = Path.GetFileName(this.FullPath).ToLower();
```

Comment:
I really wonder whether filename is enough for ID. I think there is too many possibility of collision.